database/models/events.db.php

```diff
@@ -170,6 +170,37 @@ static function listAllOpen($user_id)
         return EventDto::fromEventList($events);
     }
 
+    /** @return EventDto[] */
+    static function listAllFutureOpen($user_id)
+    {
+        $events = em()
+            ->createQuery("SELECT ev.id, ev.name, ev.start_date, ev.end_date, ev.deadline, ev.open, en.present FROM Event ev" .
+                " LEFT JOIN ev.entries en WITH en.user = ?1" .
+                " WHERE ev.open = 1" .
+                " AND ev.start_date > CURRENT_DATE()" .
```

Comment:
You might want to use end date here instead so that it's still visible until the end of the event.

Also this is almost an exact copy paste from the method below. I don't know if this is needed, but there is a possibility to reduce duplication here 🙂

Reply:
I agree for using the end_date instead of the start_date.

For the code factorisation though, I had problems when passing the boolean operator as parameter, so I'm going to leave it that way. Feel free to do it !